### PR TITLE
Repair json when LLM returns faulty responses on non json mode

### PR DESCRIPTION
**Pull request by AlonsoGuevara**
## Description
1.  This PR consolidates json string validation and loading from index, report and llm three parts into one piece.
2.  including trim extract description from llm result, replace or convert unfit chars, using json_repair for the finally repair before json loading. wrapped all functions into try_parse_json_object() as unique one function to use.     
3. There are two duplicated json_clean_up functions in both index and llm section, consolidate into one under llm model.
4. Furthermore, wrap json_clean_up into try_parse_json_object() function without Exception for friendly retry 

## Related PRs
#617 (This is a merge of this) kudos @s106916 
#665
#652 
#551 

## Proposed Changes
add json-repair

## Checklist

- [x] I have tested these changes locally.
- [x] I have reviewed the code changes.
- [x] I have updated the documentation (if necessary).
- [ ] I have added appropriate unit tests (if applicable).

## Additional Notes

[Add any additional notes or context that may be helpful for the reviewer(s).]

### CODE REVIEW

**Review by natoverse — APPROVED** [review]

**Comment by s106916** [comment]
I will close my one. :) #https://github.com/microsoft/graphrag/pull/617

**Comment by xxll88** [comment]
although json_clean_up repair json faulty responses，it increase index time and global search time 70-80%
v0.2.0 index time 35min  , v0.2.1   62min
v0.2.0 global search time  45s , v0.2.1   75s

File: graphrag/llm/openai/utils.py [hunk 1]
```diff
@@ -87,17 +89,51 @@ def get_completion_llm_args(
     }
 
 
-def try_parse_json_object(input: str) -> dict:
-    """Generate JSON-string output using best-attempt prompting & parsing techniques."""
+def try_parse_json_object(input: str) -> tuple[str, dict]:
```
Comment (jgbradley1) [inline graphrag/llm/openai/utils.py:92]: @AlonsoGuevara there is one edge case here that may explain @xxll88's comment about the significant increase in indexing/query time. Assume `input` was json-formatted correctly by the LLM - the logic in this function would still attempt to call `.replace()` multiple times on the string when it's not necessary.

Perhaps there should be a try-except call made to `json.loads(...)` at the top of this function to first check whether any further string operations are needed. Arguably, this *edge case* is the most generic case encountered when using a model that guarantee's json output. It turns into a lot of unnecessary string operations.

**Comment by MarkJGx** [comment]
No attribution to my original PR which still hasn't received a review??
https://github.com/microsoft/graphrag/pull/551

File: graphrag/llm/openai/utils.py (hunk 1 above)
Comment (AlonsoGuevara) [reply graphrag/llm/openai/utils.py:92]: Thanks for your comments @jgbradley1 and @xxll88 
Addressing this on #834

**Pull request merged**